android/app/src/main/java/labs/pooh/eaterslab/repository/dao/CafeteriaDao.kt

```diff
@@ -21,4 +21,14 @@ data class CafeteriaDao (
 
 fun Cafeteria.toDao() =
     CafeteriaDao(name, description, subDescription, longitude.toDouble(), latitude.toDouble(),
-        logoUrl, address, openedFrom, openedTo, id)
+        logoUrl, address, parseTime(openedFrom), parseTime(openedTo), id)
+
+private fun parseTime(stringTime: String): String {
```

Comment:
This should use some built-in Kotlin parser.

Reply:
not sure if  it reallu needed and available - the format is so pecific for api (as the hh:mm:ss) so it can be parsed manually - dont just see any cons of this solution

Reply:
My initial concern was that this string is a `datetime` and you try to parse it by hand.

I checked in the API definition and this value is just a string, and thus no assumptions should be taken at all. If stringTime is `"until evening"` for example, than this code will throw `IndexOutOfBoundsException` in line :30. Please remove this parsing and calls in line :24 all together.

Reply:
let's see that the parsing is done iff the not human format is detected (I mean hh:mm:ss for ex. 16:30:00) so no exception is thrown (beacuase of check `if (stringTime.count { it == ':' } > 1)`)

Reply:
sorry for the PR size - cause by the presentation and the need of refactoring the app structure in separate PRs

Reply:
Still, there may be plenty of incorrect formats, please don't make any assumptions:
* `Daily: 8-18`
* `Weekdays: 8-18, Weekends: 10-14`

It could be virtually anything. Please remove the call.

Reply:
So how to display the time in app? In the for it comes from the server? (It looks strange to me to see that the cafeteria is opened from 8:00:00 to 16:00:00). On the other hand from the server point of view these fields are datetime so they are not gonna get the form you presented (I'm not sure why API decides to serve this in this strange format). I think I can remove parsing for know and try to configure the server

Reply:
It's *not* time! It's a string, which could be some time format, but it's not guaranteed.

```
definitions:
  Cafeteria:
      ...
      opened_from:
        title: Opened from
        type: string
      ...
```

This is any `string` until API definition changes, for now it could be even crap emoji. It's server's job to expose data correct under the contract. Android client is just a presentation layer for what comes from the server.

All the comments from previous PR apply. Multiple clients, multiple API versions, two-way dependency apply here as well. Either fix it here or make a PR to add `RegexValidator` to this field so `api.yaml` adds a `pattern` to this field:


```
definitions:
  Cafeteria:
      ...
      opened_from:
        title: Opened from
        type: string
        pattern: pattern: '^\d\d:\d\d:\d\d$'
      ...
```